### PR TITLE
Travis: update build flow

### PR DESCRIPTION
**Pull request by amiclaus**
Add static analysis via cppcheck tool and code style check via astyle tool to the Travis CI build flow on m1k-fw.

Signed-off-by: Antoniu Miclaus <antoniu.miclaus@analog.com>

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -8,6 +8,11 @@ matrix:
       dist: xenial
       env:
         - BUILD_TYPE=m1k_make
+    - compiler: "gcc"
+      os: linux
+      dist: xenial
```
Comment (commodo) [inline .travis.yml]: remove `xenial` now; it's default

**Comment by amiclaus** [comment]
V2:

- remove `xenial` distribution from build matrix

**Comment by commodo** [comment]
maybe in a new commit series, treat-warnings as errors and fix the warnings somehow?
maybe with some  #ifdef magic ?

see
https://travis-ci.org/github/analogdevicesinc/m1k-fw/jobs/730261072#L430

**Comment by commodo** [comment]
curios: 
https://travis-ci.org/github/analogdevicesinc/m1k-fw/jobs/730261073

are you sure this doesn't produce any warnings/finds?

**Review by commodo — APPROVED** [review]

**Comment by commodo** [comment]
for now, this looks good as-is
maybe more things to enhance/look-at

**Pull request merged**